src/parser.y

```diff
@@ -759,10 +759,17 @@ void handle_assignment(const std::string token, Expression *expr, const Location
 
 bool parse(SourceFile *&file, const std::string& text, const std::string &filename, const std::string &mainFile, int debug)
 {
-  fs::path filepath = fs::absolute(fs::path(filename));
-  mainFilePath = fs::absolute(fs::path(mainFile));
-  parsingMainFile = mainFilePath == filepath;
+  fs::path filepath;
+  try {
+    filepath = fs::absolute(fs::path(filename));
+    mainFilePath = fs::absolute(fs::path(mainFile));
+  } catch (...) {
```

Comment:
Is there a more specific exception we can listen for?
Not a big deal, as I don't expect there to be many other failure cases, and oom errors are probably not expected to occur here.

Reply:
std::exception should work, but I wanted to be safe and catch everything.

Reply:
std::exception doesn't really make a difference IMO, I was more wondering if there's an actual subtype being thrown for the permission error. But I'm fine keeping it like this, we probably have bigger fish to fry :)